presto-iceberg/pom.xml

```diff
@@ -456,6 +456,11 @@
             <scope>provided</scope>
         </dependency>
 
+        <dependency>
+            <groupId>com.facebook.presto</groupId>
+            <artifactId>presto-parser</artifactId>
+        </dependency>
```

Comment:
Do we really need to rely on presto-parser to use a simple function logic? As I see, other connectors all do not rely on it in non-test scope.

Reply:
I went back and forth on this in my head and opted for the parser, I don't have strong feelings if you think it is not worth the dependency.

The usage is [here](https://github.com/prestodb/presto/pull/23201/files#diff-f0bfc4d32d757c7653248b29fa4f0e8656e0216c3082567cceca761393c0bcabR538) for anyone else reading this. Basically I want to ensure the SQL function is valid for all column references in the function arguments. If a column has certain characters, it's supposed to be quoted. I'm not familiar enough with the SQL specification to know if all identifiers should just be simply quoted, or if there are other rules which come into play (e.g. escaping quotes which exist in the column name, if that's even allowed).

I opted to just use the existing Presto functionality rather than trying to roll my own as I felt it was safer. At the end of the day though, the SQL spec is probably not going to change the behavior around identifier quotations, so it could be safe enough to just copy the logic in this function directly since its just a one-liner, but it just feels a little wrong to me.

Reply:
Maybe we can define a function `formatIdentifier(...)` in a certain interface of `presto-spi`, for example in `RowExpressionService` or a newly created interface, implement it in presto-main and injected to Iceberg connector in runtime. In this way, we can actually invoke the function in presto-parser while not rely on it in maven.

Reply:
Sure, I'm fine about the copying solution as the logic is so simple and clear.

BTW, SPI do not need to depend on parser in the way described above, it just need to declare a method like `String formatIdentifier(String identifier)` in some interface, like `RowExpressionService`. And the actual implementation would be in `presto-main` which depends on the parser.

Reply:
Thanks, it makes more sense now. I'll opt to leave it as is though. I do like that solution if we end up needing more functionality in the future